andela/metis-ah
# #161114854  Create CRUD operation for category resource

#### What does this PR do?
Create CRUD operation for category resource

#### Description of Task to be completed?

- modify the database relationship
- implement the functionality to access all categories
- create CRUD endpoints for categories
- add Admin role to create, delete, and update endpoints
- fetch articles from a category
- implement pagination

#### How should this be manually tested?

- Clone the repository.
- Check out the `ft-create-CRUD-operation-for-category-resource-161114854`  branch.
- Run `npm install` on your local machine.
- Run the db migrations using `npm run migrate`.
- Start the server using `npm run start_dev`.
- Using postman or any similar. get  a token upon registration and use it to test these routes:
**GET** `localhost:8000/api/v1/categories` to get all cateogries
**POST** `localhost:8000/api/v1/categories` to create a category
**GET** `localhost:8000/api/v1/categories/:categoryId` to get a single Category
**PUT** `localhost:8000/api/v1/categories/:categoryId` to edit a category
**DELETE** `localhost:8000/api/v1/categories/:categoryId` to delete a category

#### Any background context you want to provide?
This feature will enable a user to view categories and view articles under a category

#### What are the relevant pivotal tracker stories?
#161114854
